src/main/java/com/google/finapp/SpannerDaoInterface.java

```diff
@@ -25,7 +25,8 @@
   /**
    * Inserts a new row to the Customer table in the database.
    */
-  void createCustomer(ByteArray customerId, String name, String address) throws SpannerDaoException;
+  ByteArray createCustomer(ByteArray customerId, String name, String address)
```

Comment:
The interface is a bit weird: Why do we pass the customerId and also return it? The caller already knows it.

We should either:
1. Do not change the interface. The service creates one and passes it to DAO and the response.
2. Change the interface to return the Id, but then do not pass the Id as a parameter and the DAO generates it and returns it.

I think I'd prefer 1 as the UUID generation does not change based on the DAO choice.

Thoughts?

Reply:
++ on option 1

Reply:
yes, that makes sense! i think i misunderstood where to return the UUID, haha, ill change this back.